src/map/map.ts

```diff
@@ -522,18 +567,185 @@ export class PropertiesMap {
                 // same as above regarding update of the points color
                 'coloraxis.colorscale': this._options.colorScale(),
             } as unknown) as Layout);
+            this._relayout(({
+                'coloraxis2.cmax': max,
+                'coloraxis2.cmin': min,
+                // same as above regarding update of the points color
+                'coloraxis2.colorscale': this._options.colorScale(),
+            } as unknown) as Layout);
         };
 
         // ======= color palette
         this._options.palette.onchange = () => {
             this._relayout(({
                 'coloraxis.colorscale': this._options.colorScale(),
             } as unknown) as Layout);
+            this._relayout(({
+                'coloraxis2.colorscale': this._options.colorScale(),
+            } as unknown) as Layout);
+        };
+
+        // ======= opacity
+        this._options.opacity.mode.onchange = () => {
+            if (this._options.opacity.mode.value === 'filter') {
+                this._options.opacity.enableColors.enable();
```

Comment:
I'm not sold on using a lot of select/checkboxes as the user interface for this feature.

I think it is worth exploring something like a textbox/small code editor in which the user would enter mathematical expressions, something like

```
properties["PCA[1]"] < -0.2 && sin(properties["PCA[2]"]) > 0.2
```

Then we could take this expression and evaluate it for each point to decide whether to have it as opaque or transparent. This would replace `_filterForOpacity`.